darts/preprocessing/pipeline.py

```diff
@@ -0,0 +1,248 @@
+"""
+Pipeline
+--------
+"""
+from copy import deepcopy
+from collections import defaultdict
+from typing import List, Optional, Union, Tuple, Iterator
+
+from darts.logging import raise_if_not, raise_if, get_logger
+from darts import TimeSeries
+from darts.preprocessing.base_transformer import BaseTransformer
+
+logger = get_logger(__name__)
+
+
+class Pipeline:
+    def __init__(self,
+                 transformers: List[BaseTransformer[TimeSeries]],
+                 names: Optional[List[str]] = None,
+                 deep: bool = False):
+        """
+        Pipeline combines multiple transformers chaining them together.
+
+        Parameters
+        ----------
+        transformers
+            List of transformers.
+        names
+            Optional list of names for transformers must have same length as transformers and names should be unique.
+            If no names are specified transformer name will be set to str value of his index in original list.
+        deep
+            If set prevents changes in original transformers made by pipeline.
+        """
+
+        raise_if_not(all((isinstance(t, BaseTransformer)) for t in transformers),
+                     "transformers should be objects deriving from BaseTransformer", logger)
+        if deep:
+            self._transformers = deepcopy(transformers)
+        else:
+            self._transformers = transformers
+
+        if self._transformers is None or len(self._transformers) == 0:
+            self._transformers: List[BaseTransformer[TimeSeries]] = []
+            logger.warning("Empty pipeline created")
+
+        if names:
+            raise_if_not(len(names) == len(self._transformers), "names should have same length as transformer", logger)
+            raise_if_not(len(names) == len(set(names)), "names should be unique", logger)
+        else:
+            names = list(map(str, range(len(self._transformers))))
+
+        self._name_2_idx = {
+            str(name): idx for idx, name in enumerate(names)
```

Comment:
Name property of transformer? It might be okay but what about situation when same couple transformers share same name? names could be removed and we could just use indexes to simplify it

Reply:
That was my question more or less - I was wondering if it's possible to drop name property as they don't add many usability (we can name the class for that) and run on just indexing

Reply:
We could do that, it would be simpler.